apps/fyle/tasks.py

```
@@ -165,16 +161,40 @@ def create_expense_groups(
 
+def check_interval_and_sync_dimension(workspace_id: int):
+    """
+    Check sync interval and sync dimension
+    :param workspace_id: Workspace ID
+    """
+    workspace = Workspace.objects.get(id=workspace_id)
+
+    if workspace.source_synced_at:
+        time_interval = datetime.now(timezone.utc) - workspace.source_synced_at
+
+    if workspace.source_synced_at is None or time_interval.days > 0:
```

Pylint (3.3.7)

[error] Possibly using variable 'time_interval' before assignment

(E0606)